packages/cli/src/deploy/getLibraryMap.ts

```diff
@@ -0,0 +1,12 @@
+import { Hex } from "viem";
+import { Library } from "./common";
+
+export type LibraryMap = { [key: string]: Library & { address?: Hex } };
+
+export function getLibraryMap(libraries: readonly Library[]): LibraryMap {
+  return Object.fromEntries(libraries.map((library) => [getLibraryKey(library), library]));
+}
+
+export function getLibraryKey({ path, name }: { path: string; name: string }): string {
+  return `${path}:${name}`;
```

Comment:
should we include the deployer in the key since it changes the bytecode and thus the address?

Reply:
the deployer address is passed in later (in `prepareDeploy`) while the map is created before, but we can change `address` in the map to `{ [deployer: Address]: Address }` to cache per deployer address